### DISCUSSION TITLE
Wayland support is here (experimental)

### DISCUSSION THREAD

**Discussion by nbolton** [deskflow/deskflow#7456] [Announcements]
# Wayland support

Known bugs: https://github.com/deskflow/deskflow/discussions/7499

Options for enabling Wayland support:

1. [Download Deskflow v1.17](https://github.com/deskflow/deskflow/releases/tag/v1.17.0)
2. [Download Synergy v1.16](https://symless.com/synergy/download/other) (if you purchased Synergy)

Only needed if you want to tinker with the code and you're using Ubuntu: Follow the [Static `libportal` build instructions](https://github.com/deskflow/deskflow/discussions/7499#discussioncomment-11027553)

# Previous updates

**Status:** We're currently waiting for `libportal` 0.8.0 to be adopted by newer distros such as:
- Arch Linux: https://archlinux.org/packages/extra/x86_64/libportal/
- Debian 13: https://packages.debian.org/trixie/libportal1
- Ubuntu 24.04: https://packages.ubuntu.com/noble/libportal1
- Fedora 40: https://packages.fedoraproject.org/pkgs/libportal/libportal/

Special thanks to Peter Hutterer (@whot) and Olivier Fourdan (@ofourdan) from Red Hat for their work on implementing Wayland support. Special thanks to @GeorgesStavracas for releasing `libportal` 0.8.0.

## Update on 3rd Sept 2024:

Our `master` branch (since PR #7469) now makes it a _little_ easier to build the latest `libportal` version so that you can test full Wayland support (while we wait for `libportal` 0.8.0 to be adopted by Fedora 40, Debian 13, Ubuntu 24, etc.)

## Update on 2nd Sep 2024:

libportal 0.8.0 is out! 🚀

https://github.com/deskflow/deskflow/discussions/7456#discussioncomment-10521843

It's now only a matter of time before this begins making its way onto distros, which will enable Wayland support for Deskflow/Synergy.

## Original post on 30th Nov 2024:

We're currently waiting on an update from [`libportal`](https://github.com/flatpak/libportal) (the last update was about a year ago). 

~~Blocking issue: https://github.com/flatpak/libportal/issues/152~~

The latest version in their Git repo has input capture support (for the server), but the released version (0.7.1) only has remote desktop support (for the client). Once the next version of `libportal` is released, Deskflow/Synergy will have full Wayland support! :rocket: 

Note: It is possible to test Deskflow/Synergy Wayland support for both server and client, but you have to build and install the latest version of `libportal` yourself (instructions can be found on PR https://github.com/deskflow/deskflow/pull/7449).

You will need GNOME 46 or higher to use Wayland support in Deskflow/Synergy, since only this version has all the bits in mutter to make EI (emulated input) work. Right now, the distros that we officially support which this version of GNOME are:

- Debian 13
- Ubuntu 24
- Fedora 39 & 40
- OpenSUSE
- Arch Linux
- Manjaro

**Comment by nbolton** [comment]
Update: I'm currently working on making it easier for community users to build and link the latest `libportal` to enable full Wayland support, as right now the instructions in PR https://github.com/symless/synergy/pull/7449 are a bit tricky.

**Comment by atrauzzi** [comment]
Just a note, but Fedora 41 is just around the corner and I'll probably be pulling the trigger on an upgrade within the next few weeks.

**Comment by nbolton** [comment]
Alright people! It's happening! 

https://github.com/flatpak/libportal/issues/152#issuecomment-2324877664

`libportal` 0.8.0 is out! :rocket: 

Special thanks to @GeorgesStavracas

**Comment by WardBrian** [comment]
I don’t know if Synergy v3 is using the same underlying stack, can I also use Wayland with that now?

**Reply by nbolton** [reply]
> I don’t know if Synergy v3 is using the same underlying stack, can I also use Wayland with that now?

Under the hood, Synergy 3 uses the Core part of Synergy 1, and we plan on updating the Core to v1.16 in Synergy 3 to enable Wayland support in the next release of Synergy 3 (v3.2). 

However, we plan on releasing a Synergy 1 Community Edition update (v1.16) before the next Synergy 3 update. I don't have an ETA on the next Synergy 3 update at the moment.

**Reply by WardBrian** [reply]
Great!

Is there a good place to subscribe to updates for Synergy 3? Thus far I've been subscribing to the issues/discussions on this repo

**Reply by nbolton** [reply]
We plan on emailing customers to notify them about the next S1 and S3 releases, so you should get notified by email assuming you have not opted-out of account update emails.

**Comment by nbolton** [comment]
Update: Our master branch (since PR https://github.com/symless/synergy/pull/7469) now makes it a little easier to build the latest libportal version so that you can test full Wayland support (while we wait for libportal 0.8.0 to be adopted by Fedora 40, Debian 13, Ubuntu 24, etc.). See OP for test instructions.

**Comment by prestonr83** [comment]
Trying this on Arch

```
sudo pacman -S python-dbusmock python-pytest vala protobuf-c protobuf gtk3 glib2-devel gtk3 protobuf-c systemd gobject-introspection meson pugixml

meson setup build/meson -Dsystem_libportal=false
cmake -B build -DSYSTEM_LIBPORTAL=OFF

```

Getting following error even though meson build worked and the libportal.so is there.

```
$ cmake -B build -DSYSTEM_LIBPORTAL=OFF
-- Enabling warnings as errors (GNU/Clang)
-- Version number (semver): 1.16.0-beta
-- Version number (DEB/RPM): 1.16.0~beta
-- Version number (4-part): 1.16.0.0
-- Product name: Synergy 1 Community Edition
-- Disabling debug build
-- libei version: 1.3.0
CMake Warning at cmake/Libraries.cmake:246 (message):
  Local libportal not found
Call Stack (most recent call first):
  cmake/Libraries.cmake:172 (configure_libportal)
  cmake/Libraries.cmake:110 (configure_wayland_libs)
  cmake/Libraries.cmake:8 (configure_unix_libs)
  CMakeLists.txt:28 (configure_libs)


-- Qt version: 6.7.2
-- Code coverage is disabled
-- Using system GoogleTest
-- Configuring Linux packaging
-- Configuring done (0.1s)
-- Generating done (0.1s)
-- Build files have been written to: /home/preston/Documents/synergy/synergy/build
$ ls build/meson/subprojects/libportal/libportal
libportal-gtk3.so  libportal-gtk3.so.1  libportal-gtk3.so.1.0.0.p  libportal-gtk4.so  libportal-gtk4.so.1  libportal-gtk4.so.1.0.0.p  libportal-qt5.so  libportal-qt5.so.1  libportal-qt5.so.1.0.0.p  libportal.so  libportal.so.1  libportal.so.1.0.0.p
```

**Reply by prestonr83** [reply]
Not sure why but I had to build libportal subproject manually directly then copy all of the build to the meson directory and run the cmake then I was good

```
 ldd build/bin/synergys | grep libportal
	libportal.so.1 => /home/preston/Documents/synergy/synergy/build/meson/subprojects/libportal/libportal/libportal.so.1 (0x00007afdb4a69000)
```

But alas I am on hyprland and get this error when running
```
ERROR: failed to initialize remote desktop session: GDBus.Error:org.freedesktop.DBus.Error.InvalidArgs: No such interface “org.freedesktop.portal.RemoteDesktop”
```

Looks like org.freedesktop.portal.RemoteDesktop isn't quite implemented yet in xdg-desktop https://github.com/emersion/xdg-desktop-portal-wlr/pull/263

Bummer was so excited when I thought I could get this to work with hyprland

**Reply by nbolton** [reply]
Oh no! Well thanks for trying. What do you get when you run these commands?

```
$ busctl --user get-property org.freedesktop.portal.Desktop /org/freedesktop/portal/desktop org.freedesktop.portal.InputCapture  version
u 1
$ busctl --user get-property org.freedesktop.portal.Desktop /org/freedesktop/portal/desktop org.freedesktop.portal.RemoteDesktop  version
u 2
```

You should get `u 1` and `u 2` respectively. I'm guessing you won't get anything, which is sad.

Oh, and you may need to pass `--wipe` to `meson` to blow away whatever it cached previously (but I'm new to meson so take that with a pinch of salt).
```
meson setup build/meson -Dsystem_libportal=false --wipe
```

...and you'll need to tell `meson` to compile:
```
meson compile -C build/meson
```

Edit: I realised just now that I missed that from the instructions (fixed).

**Reply by nbolton** [reply]
@whot ^^

**Reply by whot** [reply]
Never used `--wipe` but thanks for the tip, should make things easier to just reconfigure. I usually just `rm -rf $builddir` :)

And yes, `meson setup` is the equivalent to autotool's `configure`, `meson compile` is the equiv to `make` (plus `meson install`, `meson test`, etc.).


The help output also says `meson configure --clearcache $builddir` exists, so that should work too.

**Reply by nbolton** [reply]
Good news, now that Arch Linux has `libportal` 0.8.1, you won't need to build against the local libportal.

@prestonr83 Would you mind opening a [bug report](https://github.com/symless/synergy/issues/new/choose) for the error you're seeing on Hyprland/xdg-desktop?

**Comment by GeorgesStavracas** [comment]
Heads up, I just released libportal 0.8.1 with some improvements and fixes for things found while releasing 0.8.0: https://github.com/flatpak/libportal/releases/tag/0.8.1

AFAICS this release won't impact Synergy, but please consider pointing to the 0.8.1 tag in the Meson subproject.

**Reply by nbolton** [reply]
> 0.8.1 tag in the Meson subproject.

Thanks for the heads up!

Perhaps a bit risky, but since the submodule is not the default, I decided on using `main` in the [Meson subproject wrap](https://github.com/symless/synergy/blob/master/subprojects/libportal.wrap) to make it easier to test bleeding edge changes. By default, our build uses system `libportal` (which is turned off with `-Dsystem_libportal=false` on `meson` and `-DSYSTEM_LIBPORTAL=OFF` on `cmake`).

**Comment by slim-salm** [comment]
Oh wow, that is indeed good news. I hope then that LMDE 7 will also receive its TLC and work out of the box. Will patiently wait and see how things develop. Thank you to all you amazing developers that make amazing tools for everyday use

**Comment by nbolton** [comment]
It's here! Synergy 1 (`v1.16.0-beta`) is now out and has experimental Wayland support.
[Arch Linux now has `libportal` 0.8.1](https://archlinux.org/packages/extra/x86_64/libportal/) and [other distros](https://pkgs.org/search/?q=libportal) will follow soon.
Discussion moved to: https://github.com/symless/synergy/discussions/7500

**Comment by atrauzzi** [comment]
Why 1 and not 2?

Sorry I'm a bit behind.

**Reply by nbolton** [reply]
Ah, Synergy 2 was discontinued. Free upgrades were given out to Synergy 2 customers for Synergy 3. I guess you missed that email. If you check your account, you should see Synergy 3 license. Synergy 3 uses Synergy 1 (open source) as its core, which is why we will maintain both versions.

**Reply by atrauzzi** [reply]
What caused 2 to be abandoned and why was 1 picked up presumably after something better enough to be 2 had already made it out? 

Just curious 🙂

**Reply by nbolton** [reply]
So many reasons: [Synergy 2: Mistakes were made](https://symless.com/synergy/news/synergy-2-mistakes-were-made)

Synergy 1 is the open source core of Synergy 3 (you can think of Synergy 3 is a pretty GUI around Synergy 1). So we need both products to continue development. Synergy 3 is proprietary for now, but I'm still considering if it should be open source. Money and customers make things complicated.

**Comment by enzious** [comment]
Just installed this on Gentoo with some slightly modified ebuild files, it was a pretty dang smooth process.

Just wanted to say THANK YOU to all involved. This was the last thing that was keeping me on X and has made my week.

**Reply by nbolton** [reply]
That's awesome! Credit mainly goes to Peter Hutterer (@whot), Olivier Fourdan (@ofourdan), and Povilas Kanapickas (@p12tic).

**Reply by enzious** [reply]
Just a note, it seems copy-paste transfer doesn't work, but that may just be the server (being Wayland) or KDE.

(moved to #7500)

**Comment by wturnerharris** [comment]
The rename and re-org caught me by surprise. Had all these 404s on open Chrome tabs for this discussion and started to freak out. 

I did end up building synergy on Linux since libportal1 is up to version 0.8.x on Ubuntu. Looks like the build worked just fine, but I'm having trouble connecting `WARNING - unrecognised client name "pc-name", check server config` on the server, but the server will not let me add the IP address of the client machine.

**Reply by nbolton** [reply]
> The rename and re-org caught me by surprise. Had all these 404s on open Chrome tabs for this discussion and started to freak out.

Sorry, I buggered up the redirect from `synergy` to `deskflow` by creating a project in it's place, ironically to help people find the new project. It looks like the redirect from `synergy-core` to `deskflow` still works, and since most links still point to `synergy-core`, I think most redirects should still work.

> WARNING - unrecognised client name "pc-name", check server config on the server, but the server will not let me add the IP address of the client machine.

Thankfully this is not Wayland related, just a typical Synergy error which happens when your client isn't in your server config. You should have gotten a dialog in the GUI asking if you want to add the client to the server config. Did you not get this prompt?

**Reply by wturnerharris** [reply]
Thanks for confirming this is likely config-specific and unrelated to Wayland. Previously with the old version, Synergy did prompt, but now it's no longer auto-discovering and when I manually add via IP or hostname, it silently fails; e.g. "continue and authorize" does nothing. I've disabled UFW, am still able to SSH and visit other web traffic/port combos I have going to that machine, so I don't know why it would be getting blocked. Thought maybe the TLS encryption could be it since Linux is using lower bit (1024 and locked as a setting on Linux)

Anyway, happy to move this over to a support thread since it builds fine with wayland support, just no way to verify it's working. Trying to help test here.

**Reply by nbolton** [reply]
Thanks for the detailed info. It sounds like we’re dealing with a network/config issue rather than something specific to Wayland. Seeking help from the Synergy support team sounds like a good idea so they can dig deeper into the details. Appreciate you helping us test!

**Reply by wturnerharris** [reply]
Just closing the loop here since I started out here. Your support team got back to me with this:
> - Firstly, unfortunately at present you can't mix Synergy 3 with Synergy 1 (or Deskflow), as these deploy different technologies for connecting (direct vs. using mDNS)
> - The logs show a config issue, and that's because Synergy 3 automatically adds the computers via synergy-service (does not exist in Synergy 1), and therefore the Linux computer is 'unrecognized' in Synergy 3. The core in Synergy 1 will try to connect, but the Synergy 3 setup will reject the connection request, as that computer is not added to the server configuration file.

I had made the assumption that since Synergy core was part of Synergy 3 that my Linux CE version would still connect with my Mac versions both running 3. Doesn't seem like that is the case. Waiting to hear back if there's a thread to follow on whether that support will come in the future or if there's a way to manually configure.

**Reply by nbolton** [reply]
There are plans to add legacy support to Synergy 3 so you can mix it together with Synergy 1/Deskflow/Input Leap/Barrier, but it might be some way off yet. Synergy 3 adds an extra network (which uses mDNS) layer that makes it incompatible. In Synergy 3, I think we need to let people turn that off so they don't have to use it and can configure the old-fashioned way.

**Comment by gabeklavans** [comment]
[Fedora 40 got libportal 0.8.x](https://packages.fedoraproject.org/pkgs/libportal/libportal/fedora-40-updates.html)! Exciting

**Reply by nbolton** [reply]
> [Fedora 40 got libportal 0.8.x](https://packages.fedoraproject.org/pkgs/libportal/libportal/fedora-40-updates.html)! Exciting

Yes it did! 🎉

**Comment by jemc** [comment]
This isn't working for me, with latest commit (6af331bd932049be33512039196273451a914fd3)

EDIT: This is resolved - see comments thread

---

OS: Ubuntu 24.04 LTS

Steps followed:

## 1. Install deps

```sh
./scripts/install_deps.sh --meson-no-system libportal --subprojects
```

Full output of that command is pasted [here](https://gist.github.com/jemc/d15d9f065041f87305f229e2ab7469c8). I did notice some red "NO"s pop up for dependencies not found. Those are pasted individually below:

```
Run-time dependency cli11 found: NO (tried pkgconfig and cmake)
[...]
libportal| Run-time dependency gtk4 found: NO (tried pkgconfig and cmake)
[...]
libportal| qmake5 found: NO need ['>= 5', '< 6']
libportal| Run-time dependency qt5 (modules: Core, Gui, Widgets, X11Extras) found: NO (tried pkgconfig and config-tool)
[...]
libportal| Program python3 (dbus, dbusmock) found: NO
```

Not sure if those are optional dependencies, but the command itself reported that it was successful overall in installing dependencies.

## 2. CMake setup

```sh
cmake -B build -DSYSTEM_LIBPORTAL=OFF
```

Among the output, I can see that it is using the local `libportal.so`
```
-- Using local subproject libportal
-- libportal library file: /home/jemc/Documents/codex/deskflow/build/meson/subprojects/libportal/libportal/libportal.so
```

## 3. Build

```sh
cmake -B build
```

## 4. Check linked libraries

```sh
ldd build/bin/deskflow-server
```

The output does NOT show libportal. Here is the full output:

```
	linux-vdso.so.1 (0x00007ffcdbf5b000)
	libX11.so.6 => /lib/x86_64-linux-gnu/libX11.so.6 (0x00007ee92e44c000)
	libxkbfile.so.1 => /lib/x86_64-linux-gnu/libxkbfile.so.1 (0x00007ee92e425000)
	libXtst.so.6 => /lib/x86_64-linux-gnu/libXtst.so.6 (0x00007ee92e41d000)
	libXext.so.6 => /lib/x86_64-linux-gnu/libXext.so.6 (0x00007ee92e409000)
	libXinerama.so.1 => /lib/x86_64-linux-gnu/libXinerama.so.1 (0x00007ee92e404000)
	libXrandr.so.2 => /lib/x86_64-linux-gnu/libXrandr.so.2 (0x00007ee92e3f5000)
	libXi.so.6 => /lib/x86_64-linux-gnu/libXi.so.6 (0x00007ee92e3e2000)
	libQt6DBus.so.6 => /lib/x86_64-linux-gnu/libQt6DBus.so.6 (0x00007ee92e32d000)
	libQt6Core.so.6 => /lib/x86_64-linux-gnu/libQt6Core.so.6 (0x00007ee92de00000)
	libssl.so.3 => /lib/x86_64-linux-gnu/libssl.so.3 (0x00007ee92dd56000)
	libcrypto.so.3 => /lib/x86_64-linux-gnu/libcrypto.so.3 (0x00007ee92d800000)
	libpugixml.so.1 => /lib/x86_64-linux-gnu/libpugixml.so.1 (0x00007ee92dd1a000)
	libnotify.so.4 => /lib/x86_64-linux-gnu/libnotify.so.4 (0x00007ee92e31d000)
	libgobject-2.0.so.0 => /lib/x86_64-linux-gnu/libgobject-2.0.so.0 (0x00007ee92d79d000)
	libstdc++.so.6 => /lib/x86_64-linux-gnu/libstdc++.so.6 (0x00007ee92d400000)
	libm.so.6 => /lib/x86_64-linux-gnu/libm.so.6 (0x00007ee92d6b4000)
	libgcc_s.so.1 => /lib/x86_64-linux-gnu/libgcc_s.so.1 (0x00007ee92d687000)
	libc.so.6 => /lib/x86_64-linux-gnu/libc.so.6 (0x00007ee92d000000)
	libxcb.so.1 => /lib/x86_64-linux-gnu/libxcb.so.1 (0x00007ee92d3d7000)
	libXrender.so.1 => /lib/x86_64-linux-gnu/libXrender.so.1 (0x00007ee92e30f000)
	libdbus-1.so.3 => /lib/x86_64-linux-gnu/libdbus-1.so.3 (0x00007ee92d388000)
	libicui18n.so.74 => /lib/x86_64-linux-gnu/libicui18n.so.74 (0x00007ee92cc00000)
	libicuuc.so.74 => /lib/x86_64-linux-gnu/libicuuc.so.74 (0x00007ee92c800000)
	libglib-2.0.so.0 => /lib/x86_64-linux-gnu/libglib-2.0.so.0 (0x00007ee92d23f000)
	libz.so.1 => /lib/x86_64-linux-gnu/libz.so.1 (0x00007ee92d223000)
	libdouble-conversion.so.3 => /lib/x86_64-linux-gnu/libdouble-conversion.so.3 (0x00007ee92cfeb000)
	libb2.so.1 => /lib/x86_64-linux-gnu/libb2.so.1 (0x00007ee92cfcd000)
	libpcre2-16.so.0 => /lib/x86_64-linux-gnu/libpcre2-16.so.0 (0x00007ee92cb74000)
	libzstd.so.1 => /lib/x86_64-linux-gnu/libzstd.so.1 (0x00007ee92caba000)
	/lib64/ld-linux-x86-64.so.2 (0x00007ee92e74c000)
	libgdk_pixbuf-2.0.so.0 => /lib/x86_64-linux-gnu/libgdk_pixbuf-2.0.so.0 (0x00007ee92cf9f000)
	libgio-2.0.so.0 => /lib/x86_64-linux-gnu/libgio-2.0.so.0 (0x00007ee92c630000)
	libffi.so.8 => /lib/x86_64-linux-gnu/libffi.so.8 (0x00007ee92d217000)
	libXau.so.6 => /lib/x86_64-linux-gnu/libXau.so.6 (0x00007ee92d681000)
	libXdmcp.so.6 => /lib/x86_64-linux-gnu/libXdmcp.so.6 (0x00007ee92cf97000)
	libsystemd.so.0 => /lib/x86_64-linux-gnu/libsystemd.so.0 (0x00007ee92c550000)
	libicudata.so.74 => /lib/x86_64-linux-gnu/libicudata.so.74 (0x00007ee92a600000)
	libpcre2-8.so.0 => /lib/x86_64-linux-gnu/libpcre2-8.so.0 (0x00007ee92ca20000)
	libgomp.so.1 => /lib/x86_64-linux-gnu/libgomp.so.1 (0x00007ee92c4f9000)
	libgmodule-2.0.so.0 => /lib/x86_64-linux-gnu/libgmodule-2.0.so.0 (0x00007ee92cf90000)
	libpng16.so.16 => /lib/x86_64-linux-gnu/libpng16.so.16 (0x00007ee92cf58000)
	libjpeg.so.8 => /lib/x86_64-linux-gnu/libjpeg.so.8 (0x00007ee92c476000)
	libmount.so.1 => /lib/x86_64-linux-gnu/libmount.so.1 (0x00007ee92c429000)
	libselinux.so.1 => /lib/x86_64-linux-gnu/libselinux.so.1 (0x00007ee92c3fc000)
	libbsd.so.0 => /lib/x86_64-linux-gnu/libbsd.so.0 (0x00007ee92c3e6000)
	libcap.so.2 => /lib/x86_64-linux-gnu/libcap.so.2 (0x00007ee92ca13000)
	libgcrypt.so.20 => /lib/x86_64-linux-gnu/libgcrypt.so.20 (0x00007ee92a4b8000)
	liblz4.so.1 => /lib/x86_64-linux-gnu/liblz4.so.1 (0x00007ee92c3c4000)
	liblzma.so.5 => /lib/x86_64-linux-gnu/liblzma.so.5 (0x00007ee92c392000)
	libblkid.so.1 => /lib/x86_64-linux-gnu/libblkid.so.1 (0x00007ee92a47d000)
	libmd.so.0 => /lib/x86_64-linux-gnu/libmd.so.0 (0x00007ee92c383000)
	libgpg-error.so.0 => /lib/x86_64-linux-gnu/libgpg-error.so.0 (0x00007ee92a458000)
```

## 5. Try to Run

When trying to run `build/bin/deskflow` in server mode, I see the Wayland warning:

> Sorry, while this version of Deskflow does support Wayland, this build was not linked with one or more of the required libraries

After accepting the warning the app tries repeatedly to run the server and fails, with logs like this in a loop:

```
[2024-10-15T08:34:58] INFO: starting core server process (desktop mode)
[2024-10-15T08:34:58] WARNING: missing required wayland lib(s) or feature
[2024-10-15T08:34:58] INFO: core config file: /home/jemc/.config/Deskflow/deskflow-server.conf
[2024-10-15T08:34:58] INFO: running command: /home/jemc/Documents/codex/deskflow/build/bin/deskflow-server -f --no-tray --debug INFO --name jemc-f16 --enable-crypto --address :24800 -c /home/jemc/.config/Deskflow/deskflow-server.conf --tls-cert /home/jemc/.config/Deskflow/tls/deskflow.pem
[2024-10-15T08:34:58] WARNING: desktop process exited with error code: 1
```

If I try to run the same server command myself from a terminal, I can see the error that causes it to exit in failure:

```
/home/jemc/Documents/codex/deskflow/build/bin/deskflow-server -f --no-tray --debug INFO --name jemc-f16 --enable-crypto --address :24800 -c /home/jemc/.config/Deskflow/deskflow-server.conf --tls-cert /home/jemc/.config/Deskflow/tls/deskflow.pem
```
```
[2024-10-15T08:35:41] FATAL: a runtime error occurred: libei is not supported
```

**Reply by nbolton** [reply]
Hmm, seems like missing `libei` is the issue. Could you clear your CMake cache and share the full config output of `cmake -B build -DSYSTEM_LIBPORTAL=OFF`?

**Reply by jemc** [reply]
Thanks. Building `libei` from source and installing it (then rebuilding deskflow) fixed the issue. :tada: 

I guess I had assumed that either `scripts/install_deps.sh` would handle that for me, or that I'd get a CMake configuration error if I was missing the `libei`.

**Reply by nbolton** [reply]
Yes, my fault. I had hoped that the script would make it easier for contributors but that has backfired. 

Currently, @sithlord48 is planning to remove `scripts/install_deps.sh` as it seems to be causing more harm than good.

**Reply by sithlord48** [reply]
all in due time

**Comment by OldManBrodie** [comment]
So I updated to 24.10 today, which switched me over to Wayland, and now Synergy won't work, so I found this page, but the instructions don't seem to work, since a number of the switches specified in the install_deps.py file were removed two days ago with commit [7b7ff81](https://github.com/deskflow/deskflow/commit/7b7ff81a9fa270cedabaea8e0b6b4360a5787ee3)

Is there a way to build this now? I'm kind of new to Linux, but I'm a developer by trade, so the instructions seem pretty straight-forward.

Or is a version of Synergy coming out any day now that supports Wayland, and I should just wait?

**Reply by nbolton** [reply]
Welcome to the Deskflow community! This is the new upstream project for Synergy.

Your options for Wayland support are:
1. [Download Synergy v1.16](https://symless.com/synergy/download/other) (if you have a license)
2. [Download Deskflow v1.17](https://github.com/deskflow/deskflow/releases/tag/v1.17.0) (no license required)
3. Follow the [newer build instructions](https://github.com/deskflow/deskflow/discussions/7499#discussioncomment-11027553) (only needed if you want to tinker with the code)

I've updated the main article to point to those.

**Reply by OldManBrodie** [reply]
I've been using Synergy 3, is this update going to make it into that soon?

**Reply by p-kostic** [reply]
This discussion pops-up as the first result in Google for the query 'synergy wayland' (once you follow a reddit comment). 

I've been wondering the same as amccarter. Symless Synergy is the only app that's keeping me from fully switching to wayland.

**Reply by nbolton** [reply]
We're working hard on trying to get 3.2 out the door right now. We keep finding small bugs that we need to fix before we can ship, so we're behind schedule. Thanks for your patience guys. I the meantime, please try [Synergy v1.16](https://symless.com/synergy/download/other) or [Deskflow v1.17](https://github.com/deskflow/deskflow/releases/tag/v1.17.0).

**Comment by nbolton** [comment]
### Old Wayland build instructions (do not use)

> [!WARNING]
> Do not use these build instructions. I'm saving these for posterity.
>
> Instead, [download Deskflow v1.17.0](https://github.com/deskflow/deskflow/releases/tag/v1.17.0) or follow the [newer build instructions](https://github.com/deskflow/deskflow/discussions/7499#discussioncomment-11027553).

1. Use GNOME46 (sorry, haven't tested with KDE yet but I plan to)
1. Build local `libportal`:
    - Run: `./scripts/install_deps.py --meson-no-system libportal --subprojects`
    - Or, install the deps for `libportal` manually and run Meson setup and compile direct:
        - `meson setup build/meson -Dsystem-libportal=false`
        - `meson compile -C build/meson`
3. Configure with system `libportal` disabled: `cmake -B build -DSYSTEM_LIBPORTAL=OFF`
4. Build the project: `cmake --build build -j8`
5. Verify that `synergys` is linked against the local `libportal.so` that you just built:
    ```
    $ ldd build/bin/synergys | grep libportal
    libportal.so.1 => /path/to/synergy/build/meson/subprojects/libportal/libportal/libportal.so.1 (0x00007f8aa1160000)
    ```
    ... and not the system `libportal` (e.g. `/usr/lib`, `/usr/local/lib`, etc)
6. Run using the GUI: `build/bin/synergy`
7. Verify that you see: `INFO: using ei screen for wayland` in the logs
